helpers.php

```diff
@@ -937,6 +1060,62 @@ function updateOrder($cart, $order, $response, $shopOrderId, $lockFileName, $loc
     }
 }
 
+/**
+ * @param $cart
+ * @param $order
+ * @param $response
+ * @param $shopOrderId
+ * @param $lockFileName
+ * @param $lockFileHandle
+ *
+ * @return void
+ */
+function updateChildOrder($cart, $order, $response, $shopOrderId, $lockFileName, $lockFileHandle)
+{
+    $module = Module::getInstanceByName('altapay');
+    if ($response && is_array($response->Transactions)) {
+        $transactionStatus = $response->Transactions[0]->TransactionStatus;
+    }
+    $auth_statuses = ['preauth', 'invoice_initialized', 'recurring_confirmed'];
+    $captured_statuses = ['bank_payment_finalized', 'captured'];
+    if (in_array($transactionStatus, $auth_statuses, true) or in_array($transactionStatus, $captured_statuses, true)) {
+        // Update payment status to 'succeeded'
+        $sql = 'UPDATE `' . _DB_PREFIX_ . 'altapay_child_order` 
+        SET `paymentStatus` = \'succeeded\' WHERE `unique_id` = ' . $shopOrderId;
+        Db::getInstance()->Execute($sql);
+        $payment = $order->getOrderPaymentCollection();
+        if (isset($payment[0])) {
+            $payment[0]->transaction_id = pSQL($shopOrderId);
+            $payment[0]->save();
+        }
+
+        if (!empty($response->Transactions[0]->ReconciliationIdentifiers)) {
+            $reconciliation_identifier = $response->Transactions[0]->ReconciliationIdentifiers[0]->Id;
+            $reconciliation_type = $response->Transactions[0]->ReconciliationIdentifiers[0]->Type;
+
+            saveChildOrderIdentifier($order->id, $reconciliation_identifier, $reconciliation_type, $shopOrderId);
+        }
+        $customer = new Customer($cart->id_customer);
+        unlockCallback($lockFileName, $lockFileHandle);
+        Tools::redirect('index.php?controller=order-confirmation&id_cart=' . (int) $cart->id . '&id_module=' . (int) $module->id . '&id_order=' . $order->id . '&key=' . $customer->secure_key);
+    } elseif ($transactionStatus === 'epayment_declined') {
+        // Update payment status to 'declined'
+        $sql = 'UPDATE `' . _DB_PREFIX_ . 'altapay_child_order` 
+            SET `paymentStatus` = \'declined\' WHERE `unique_id` = ' . $shopOrderId;
+        Db::getInstance()->Execute($sql);
+        unlockCallback($lockFileName, $lockFileHandle);
+        exit('Order status updated to Error');
+    } else {
+        // Unexpected scenario
+        $mNa = $module->name;
+        PrestaShopLogger::addLog('Unexpected scenario: Callback notification was received for Transaction '
+            . $shopOrderId . ' with payment status ' . $transactionStatus, 3, '1005', $mNa,
```

Comment:
Same comments as above.

Reply:
We are already applying the validation to these variables, as we are logging errors here.